kernel/packages/shared/comms/chat.ts

```diff
@@ -25,3 +27,41 @@ export function notifyStatusThroughChat(status: string) {
     })
   )
 }
+
+export function notifyFriendOnlineStatusThroughChat(userStatus: UpdateUserStatusMessage) {
+  const friendName = getProfile(globalThis.globalStore.getState(), userStatus.userId)?.name
+
+  if (friendName === undefined) {
+    return
+  }
+
+  if (friendStatus[friendName] === undefined) {
+    friendStatus[friendName] = userStatus.presence
+    return
+  }
+
+  if (!(userStatus.realm?.layer && userStatus.realm?.serverName)) {
+    if (userStatus.presence !== PresenceStatus.ONLINE) {
+      friendStatus[friendName] = userStatus.presence
+    }
+    return
+  }
+
+  if (userStatus.presence === PresenceStatus.ONLINE && friendStatus[friendName] === PresenceStatus.OFFLINE) {
+    let message = `${friendName} joined ${userStatus.realm?.serverName}-${userStatus.realm?.layer}`
+
+    if (userStatus.position) {
+      message += ` ${userStatus.position.x}, ${userStatus.position.y}`
+    }
+
+    globalThis.globalStore.dispatch(
+      messageReceived({
+        messageId: uuid(),
+        messageType: ChatMessageType.SYSTEM,
+        timestamp: Date.now(),
+        body: message
+      })
+    )
```

Comment:
I'd rather use the `shared/comms/chats.ts#notifyStatusThroughChat` function instead to avoid repetition.

Reply:
Agree. This could be replaced with: `notifyStatusThroughChat(message)`